core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/ExecInfoAnalyzer.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool.analysis
+
+import scala.collection.mutable
+
+import com.nvidia.spark.rapids.tool.planparser.{ExecInfo, ExecRef, ExprRef, OpTypes}
+
+case class ExecInfoAnalyzer(execInfos: Seq[ExecInfo]) {
+
+  // Internal case classes for uniquely identifying execs and expressions
+  private case class OperatorKey(nameRef: ExecRef, opType: OpTypes.OpType, isSupported: Boolean)
+  private case class ExpressionKey(nameRef: ExprRef, opType: OpTypes.OpType, isSupported: Boolean)
+
+  /**
+   * Holds aggregated data for an exec.
+   *
+   * @param count  Number of times the exec occurs.
+   * @param stages Set of stages where the exec is found.
+   */
+  case class ExecData(
+      var count: Int = 0,
+      var stages: Set[Int] = Set()
+  )
+
+  /**
+   * Holds aggregated data for an expression.
+   *
+   * @param count  Number of times the expression occurs.
+   * @param stages Set of stages where the expression is found.
+   */
+  case class ExpressionData(
+      var count: Int = 0,
+      var stages: Set[Int] = Set()
+  )
+
+  // Internal data structure for aggregating execs and expressions by SQL ID.
+  // The structure is a Map from SQL ID to a Map of OperatorKey to a tuple containing
+  // ExecData and a Map of ExpressionKey to ExpressionData.
+  private val aggregatedData: mutable.Map[Long, mutable.Map[OperatorKey,
+      (ExecData, mutable.Map[ExpressionKey, ExpressionData])]] = mutable.Map()
+
+  def analyze(): Unit = {
+    execInfos.foreach(traverse)
+  }
+
+  /**
+   * Recursively traverses the execution tree to collect exec and expression statistics
+   *
+   * @param execInfo The execution information node to process
+   */
+  private def traverse(execInfo: ExecInfo): Unit = {
```

Comment:
IS this to support future when we count expressions per exec? because for now, we only have a single exprRef that cannot be more than 1.

Reply:
That's correct. Was planning to do it as a follow-on.  Will file an issue to refactor the code further.